tests/components/zwave/test_init.py

```diff
@@ -1310,699 +1235,851 @@ def mock_connect(receiver, signal, *args, **kwargs):
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert value_list.data == "item3"
+
+    # Button
+    await hass.services.async_call(
+        "zwave",
+        "set_config_parameter",
+        {
+            const.ATTR_NODE_ID: 14,
+            const.ATTR_CONFIG_PARAMETER: 14,
+            const.ATTR_CONFIG_VALUE: True,
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert zwave_network.manager.pressButton.called
+    assert zwave_network.manager.releaseButton.called
+
+    # List of Ints
+    await hass.services.async_call(
+        "zwave",
+        "set_config_parameter",
+        {
+            const.ATTR_NODE_ID: 14,
+            const.ATTR_CONFIG_PARAMETER: 15,
+            const.ATTR_CONFIG_VALUE: 3,
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert value_list_int.data == "3"
+
+    # Boolean Truthy
+    await hass.services.async_call(
+        "zwave",
+        "set_config_parameter",
+        {
+            const.ATTR_NODE_ID: 14,
+            const.ATTR_CONFIG_PARAMETER: 16,
+            const.ATTR_CONFIG_VALUE: "True",
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert value_bool.data == 1
+
+    # Boolean Falsy
+    await hass.services.async_call(
+        "zwave",
+        "set_config_parameter",
+        {
+            const.ATTR_NODE_ID: 14,
+            const.ATTR_CONFIG_PARAMETER: 16,
+            const.ATTR_CONFIG_VALUE: "False",
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert value_bool.data == 0
+
+    # Different Parameter Size
+    await hass.services.async_call(
+        "zwave",
+        "set_config_parameter",
+        {
+            const.ATTR_NODE_ID: 14,
+            const.ATTR_CONFIG_PARAMETER: 19,
+            const.ATTR_CONFIG_VALUE: 0x01020304,
+            const.ATTR_CONFIG_SIZE: 4,
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert node.set_config_param.called
+    assert len(node.set_config_param.mock_calls) == 1
+    assert node.set_config_param.mock_calls[0][1][0] == 19
+    assert node.set_config_param.mock_calls[0][1][1] == 0x01020304
+    assert node.set_config_param.mock_calls[0][1][2] == 4
+    node.set_config_param.reset_mock()
+
+
+async def test_print_config_parameter(hass, mock_openzwave, zwave_setup):
+    """Test zwave print_config_parameter service."""
+    zwave_network = hass.data[DATA_NETWORK]
+    zwave_network.state = MockNetwork.STATE_READY
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
+    await hass.async_block_till_done()
+
+    value1 = MockValue(
+        index=12, command_class=const.COMMAND_CLASS_CONFIGURATION, data=1234
+    )
+    value2 = MockValue(
+        index=13, command_class=const.COMMAND_CLASS_CONFIGURATION, data=2345
+    )
+    node = MockNode(node_id=14)
+    node.values = {12: value1, 13: value2}
+    zwave_network.nodes = {14: node}
+
+    with patch.object(zwave, "_LOGGER") as mock_logger:
```

Comment:
A future improvement is to replace this by using the `caplog` fixture.

Reply:
Thanks! Changed that to `caplog`, wasn't too happy either with the way this ended up.